startup/10-detectors.py

```diff
@@ -96,102 +98,149 @@ class Encoder(Device):
+    # path without the root
+    filepath = os.path.join(datetime.now().strftime(write_path_template), filename)
+    return filepath
 
-print("init EncoderFS")
+
+# TODO: Move this class to ophyd.
 class EncoderFS(Encoder):
     "Encoder Device, when read, returns references to data in filestore."
     chunk_size = 2**20
-    write_path_template = '/nsls2/xf07bm/data/pizza_box_data/%Y/%m/%d/'    
-    
+    write_path_template = '/nsls2/xf07bm/data/pizza_box_data/%Y/%m/%d/'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._asset_docs_cache = deque()
+        self._resource_uid = None
+        self._datum_counter = None
+
+    def collect_asset_docs(self):
+        items = list(self._asset_docs_cache)
+        self._asset_docs_cache.clear()
+        for item in items:
+            yield item
+
     def stage(self):
         "Set the filename and record it in a 'resource' document in the filestore database."
 
-        if(self.connected):
-        #if True:
-            print(self.name, 'stage')
-            DIRECTORY = datetime.now().strftime(self.write_path_template)
-            #DIRECTORY = "/nsls2/xf07bm/data/pb_data"
-
-            filename = 'en_' + str(uuid.uuid4())[:6]
-            self._full_path = os.path.join(DIRECTORY, filename)  # stash for future reference
-            print("writing data to {}".format(self._full_path))
+        if self.connected:
+            print('Staging of {} starting'.format(self.name))
 
-            self.filepath.put(self._full_path)
-            self.resource_uid = self._reg.register_resource(
-                'PIZZABOX_ENC_FILE_TXT',
-                DIRECTORY, self._full_path,
-                {'chunk_size': self.chunk_size})
+            filename = 'en_' + str(uuid.uuid4())[:8]
+            
+            DIRECTORY = datetime.now().strftime(self.write_path_template)
+            # without the root, but with data path + date folders
+            full_path = make_filename(filename)
+            # with the root
+            self._full_path = os.path.join(DIRECTORY, full_path)  # stash for future reference
+
+
+            # FIXME: Quick TEMPORARY fix for beamline disaster
+            # we are writing the file to a temp directory in the ioc and
+            # then moving it to the GPFS system.
+            #
+            #ioc_file_root = '/home/softioc/tmp/'
+            #self._ioc_full_path = os.path.join(ioc_file_root, filename)
+            #self._filename = filename
+
+            self.filepath.put(self._full_path)   # commented out during disaster
+            #self.filepath.put(self._ioc_full_path)
+
+            self._resource_uid = str(uuid.uuid4())
+            resource = {'spec': 'PIZZABOX_ENC_FILE_TXT',
+                        'root': DIRECTORY,
+                        'resource_path': full_path,
+                        'resource_kwargs': {'chunk_size': self.chunk_size},
+                        'path_semantics': os.name,
```

Comment:
I believe it should be the following line according to https://github.com/bluesky/ophyd/pull/705/files#diff-80d06b5ac1409f465083a6abb44a9b14R714:
```suggestion
                        'path_semantics': {'posix': 'posix', 'nt': 'windows'}[os.name]}
```